src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala

```diff
@@ -92,7 +99,32 @@ class PodsController(
       }
     }
 
-  def update(podId: PathId): Route = ???
+  @SuppressWarnings(Array("all")) // async/await
+  def update(podId: PathId): Route = {
+    authenticated.apply { implicit identity =>
+      (entity(as[raml.Pod]) & forceParameter & extractClientIP & extractUri) {
+        case (ramlPod, force, host, uri) =>
+          assumeValid(podDefValidator().apply(ramlPod)) {
+            normalized(ramlPod, podNormalizer) { normalizedPodDef =>
+              val pod = Raml.fromRaml(normalizedPodDef).copy(version = clock.now())
+              assumeValid(PodsValidation.pluginValidators(pluginManager).apply(pod)) {
+                authorized(UpdateRunSpec, pod).apply {
+                  val deploymentPlan = async {
+                    val plan = await(podManager.update(pod, force))
+                    eventBus.publish(PodEvent(host.toString(), uri.toString(), PodEvent.Updated))
+                    plan
+                  }
+                  onSuccessLegacy(deploymentPlan).apply { plan =>
```

Comment:
I must be missing something. Which exception is thrown by `update` that has to be handled? `onSuccessLegacy` lists `ValidationFailedException`, `AccessDeniedException`, `AuthDirectives.NotAuthorized` which are already handled. This leaves `ConflictingChangeException`. This could be handled in the async block, no? So we would not requires the legacy implementation.